### PR TITLE
GameCommand::createTree

### PR DESCRIPTION
**Pull request by duncanspumpkin**
This is the first create game command so quite a bit of unrelated work required to get the mechanisms ready from TileManager.

### CODE REVIEW

File: src/OpenLoco/GameCommands/CreateTree.cpp [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+        {
+            setErrorText(StringIds::land_type_not_suitable);
+            return FAILURE;
+        }
+
+        const auto baseZ = quadrantHeight.landHeight / 4;
+        auto clearanceZ = baseZ + treeObj->var_02;
+        if (args.requiresFullClearance)
+        {
+            clearanceZ = 255;
```
Comment (LeftofZen) [inline src/OpenLoco/GameCommands/CreateTree.cpp:82]: any reason for using 255 instead of 0xFF or std::numeric_limits<uint8_t>::max()?

Comment (duncanspumpkin) [reply src/OpenLoco/GameCommands/CreateTree.cpp:82]: I was trying to find if we had defined it anywhere else it then forgot what i was doing.

**Comment by LeftofZen** [comment]
There is a small bug in this PR where trees are allowed to be placed on sloping underwater land that is 1 tile deep:

Current master:
image

This PR:
image

File: src/OpenLoco/GameCommands/CreateTree.cpp [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+        auto* elSurface = Map::TileManager::get(args.pos).surface();
+        if (elSurface == nullptr)
+        {
+            return FAILURE;
+        }
+
+        if (elSurface->water())
+        {
+            if (elSurface->water() * 8 - 1 > quadrantHeight.landHeight)
```
Comment (AaronVanGeffen) [inline src/OpenLoco/GameCommands/CreateTree.cpp:64]: Should this be 4 instead of 8?

Comment (duncanspumpkin) [reply src/OpenLoco/GameCommands/CreateTree.cpp:64]: no water is special :) its in microZ

**Comment by duncanspumpkin** [comment]
Hmm this is exposing a few little issues that have been there for a while. Looks like i copied pasted some buggy OpenRCT2 code (the OpenRCT2 code has since been fixed though so its just our version that was buggy).

**Comment by duncanspumpkin** [comment]
Okay that should be all the height issues sorted.

**Comment by LeftofZen** [comment]
Almost perfect but there's still an 'off-by-one' kind of bug that lets you place trees on diagonal sloping underwater tiles, on the 1 subtile that is raised, even though its fully underwater:

Master: 
image


PR:
image

**Comment by duncanspumpkin** [comment]
Urgh there was yet another little bug in the height calc code. Thats it now fixed for sure.

**Review by LeftofZen — APPROVED** [review]
All issues seem to have been resolved as far as I can test

**Pull request merged**